### PR TITLE
Fix panic on shutdown form orphaned goroutine

### PR DESCRIPTION
**Pull request by aaronbuchwald**

### CODE REVIEW

File: plugin/evm/vm.go [hunk 1]
```diff
@@ -745,6 +723,52 @@ func (vm *VM) writeBackMetadata() {
 	atomic.StoreUint32(&vm.writingMetadata, 0)
 }
 
+// awaitTxPoolStabilized waits for a txPoolHead channel event
+// and notifies the VM when the tx pool has stabilized to the
+// expected block hash
+// Waits for signal to shutdown from txPoolStabilizedShutdownChan chan
+func (vm *VM) awaitTxPoolStabilized() {
+	vm.shutdownWg.Add(1)
```
Comment (StephenButtolph) [inline plugin/evm/vm.go:731]: This should be called in the main goroutine, not the spawned goroutine.

File: plugin/evm/vm.go [hunk 2]
```diff
@@ -745,6 +723,52 @@ func (vm *VM) writeBackMetadata() {
...
+				vm.txPoolStabilizedLock.Lock()
+				if vm.txPoolStabilizedHead == h.Block.Hash() {
+					vm.txPoolStabilizedOk <- struct{}{}
+					vm.txPoolStabilizedHead = common.Hash{}
+				}
+				vm.txPoolStabilizedLock.Unlock()
+			default:
+			}
+		case <-vm.txPoolStabilizedShutdownChan:
+			vm.shutdownWg.Done()
```
Comment (StephenButtolph) [inline plugin/evm/vm.go:746]: Would it make sense for this to be in a defer at the start of the function?

File: plugin/evm/vm.go [hunk 3]
```diff
@@ -745,6 +723,52 @@ func (vm *VM) writeBackMetadata() {
...
+			return
+		}
+	}
+}
+
+func (vm *VM) awaitSubmittedTxs() {
```
Comment (StephenButtolph) [inline plugin/evm/vm.go:752]: Same comments as `awaitTxPoolStabilized`

File: plugin/evm/vm.go (hunk 2 above)
Comment (aaronbuchwald) [reply plugin/evm/vm.go]: Yup that would be much better

File: plugin/evm/vm.go (hunk 1 above)
Comment (aaronbuchwald) [reply plugin/evm/vm.go]: Updated

**Review by StephenButtolph — APPROVED** [review]

**Pull request merged**